src/api/types.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <memory>
+#include "../types/gamestate.h"
+
+namespace api {
+
+extern "C" {
+
+typedef int CPiece;
+typedef int CWind;
+
+typedef struct CEvent {
+  int type;
```

Comment:
Might be worth trying to use an enum here. We might be able to refactor the event type c++ header to be C compatible and use the same header, but otherwise we could consider using an enum here. 
But I don't know if it's needed for this largely compatibility layer, so maybe we can leave it for now and decide later if there's really a use for it.

Reply:
Yea I think once we have the proto-based models integrated we should revisit and see if either *that* can provide the enum for us or if given the compat layer each side can have the enums itself

Reply:
Yea I think once we have the proto-based models integrated we should revisit this and see if either that can provide the enum for us or if given the compat layer each side can have the enums itself

For now treating it as a thin layer feels reasonable